configure

```diff
@@ -0,0 +1,372 @@
+            envvar_value=$(expr "X$arg" : '[^=]*=\(.*\)') || envvar_value="" # Fix empty var bug
+            eval "$envvar=$envvar_value"
+        ;;
+        *=*) warning Unknown variable: "$(expr "x$arg" : 'x\([^=]*\)=')" ;;
+        *) warning Unknown argument: "$arg" ;;
+    esac
+done
+
+## General Defines
+: "${PLATFORM:=$(uname)}"
+: "${PREFIX:="/usr"}"
+: "${EPREFIX:="/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SBINDIR:="/sbin"}"
+: "${MANDIR:="/share/man/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SHUTDOWN_PREFIX:=""}"
+: "${CXXFLAGS_EXTRA:=""}"
+: "${TEST_CXXFLAGS:=""}"
+: "${TEST_CXXFLAGS_EXTRA:=""}"
+: "${CPPFLAGS:=""}"
+: "${LDFLAGS_EXTRA:=""}"
+: "${TEST_LDFLAGS_EXTRA:=""}"
+if [ "$PLATFORM" = "Linux" ]; then
+    : "${BUILD_SHUTDOWN:=yes}"
+    : "${SUPPORT_CGROUPS:=1}"
+    : "${SYSCONTROLSOCKET:="/run/dinitctl"}"
+else
+    : "${BUILD_SHUTDOWN:=no}"
+    : "${SUPPORT_CGROUPS:=0}"
+    : "${SYSCONTROLSOCKET:="/var/run/dinitctl"}"
+fi
+
+## Resolve final paths
+SBINDIR="$EPREFIX/$SBINDIR"
+MANDIR="$PREFIX/$MANDIR"
+
+## Finalize $CXXFLAGS, $LDFLAGS, $TEST_LDFLAGS
+if [ -z "${CXXFLAGS+IS_SET}" ]; then
+    CXXFLAGS=""
+    AUTO_CXXFLAGS="true"
+else
+    AUTO_CXXFLAGS="false"
+fi
+if [ -z "${LDFLAGS+IS_SET}" ]; then
+    LDFLAGS=""
+    AUTO_LDFLAGS="true"
+else
+    AUTO_LDFLAGS="false"
+fi
+if [ -z "${TEST_LDFLAGS+IS_SET}" ]; then
+    TEST_LDFLAGS=""
+    AUTO_TEST_LDFLAGS="true"
+else
+    AUTO_TEST_LDFLAGS="false"
+fi
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Known Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+create_testfile_cc
+
+## Find and test C++ compiler
+if [ -z "${CXX:-}" ]; then
+    info Checking C++ compiler...
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if [ -n "${CXX:-}" ]; then
+            sub_info "$CXX"
+            break # Found
+        fi
+    done
+    if [ -z "${CXX:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works "$CXX"
+if [ -n "${CXX_FOR_BUILD:-}" ]; then
+    cxx_works "$CXX_FOR_BUILD"
+fi
+
+## Test compiler/linker supported arguments
+if [ "$AUTO_CXXFLAGS" = "true" ]; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXFLAGS $argument
+    done
+    if [ "$PLATFORM" != "FreeBSD" ]; then
+        try_cxx_argument CXXFLAGS -flto
+    fi
+    if [ "$PLATFORM" != "Darwin" ]; then
+        try_cxx_argument CXXFLAGS -fno-rtti
+    fi
+fi
+if [ "$PLATFORM" = "FreeBSD" ] && [ "$AUTO_LDFLAGS" = "true" ]; then
+    try_ld_argument LDFLAGS -lrt
+fi
+if [ "${NO_SANITIZE:-}" = "auto" ] && [ "$AUTO_TEST_LDFLAGS" = "true" ]; then
+    try_ld_argument TEST_LDFLAGS -fsanitize=address,undefined
```

Comment:
`-fsanitize` options are needed for compilation as well, not just linking

Reply:
I think if a flag can pass `try_ld_argument()`, It can added to `$CXXFLAGS` without test it with `try_cxx_argument()`. Right?

Reply:
Yes, `try_ld_argument` is doing compilation and linking so any argument that works is usable for both.

(This doesn't extend to all arguments though. For instance we don't want to put `-lrt` on the compiler command line even if it can be used when linking.)